src/libraries/Common/src/System/Diagnostics/DiagnosticsHelper.cs

```diff
@@ -7,62 +7,241 @@ namespace System.Diagnostics
 {
     internal static class DiagnosticsHelper
     {
-        internal static bool CompareTags(IEnumerable<KeyValuePair<string, object?>>? tags1, IEnumerable<KeyValuePair<string, object?>>? tags2)
+        // This is similar to System.Linq ToArray. We are not using System.Linq here to avoid the dependency.
```

Comment:
Why is the dependency problematic?

Reply:
The auto-instrumentation can force loading higher version of `System.Diagnostics.DiagnosticSource` than the version of the one that app built with. Increasing the dependency of `System.Diagnostics.DiagnosticSource` can increase the risk of breaking the auto-instrumentation. Here is some related issue talking about similar thing https://github.com/dotnet/runtime/issues/42244